### ISSUE TITLE
Project roadmap

### ISSUE THREAD

**Issue by Amaumaury** [Amaumaury/ada-2017#2]
I am still very skeptical about the project. The best in my opinion is that we try to setup a clear roadmap asap. If we can produce one then we will have a clear idea of the project that will allow us to move fast. If we cannot produce the roadmap then we will be able to change the project right on time (or come up with another angle of approach). 

So how I see the project now is that we want to study impact of events on countries through the wikipedia edits. The intuition being that the more important the event is, then more modifications will be observed at deeper levels on the knowledge graph. 

Everything that I’m going to say from now is based on the project idea I describe above. So if my understanding is « true » then it implies the following steps for the project : 

**I - Being able to have an exhaustive characterization for any country thought a set of wikipedia pages.** 

Why is this necessary ? We cannot just focus only on the main wikipedia page of any country. They give partial information and many changes are probably not reported at this level of description.  

Brief ideas on how to do it for a specific country c : 
Perform BFS and stop at predetermined depth level. This method is not good, many false positive and false negative will happen.
Perform a special BFS on the wikipedia graph starting at the c’s main wikipedia page that stops exploring a branch **when the new paged read is not relevant to the country**. This is actually hard to do; we could use some text processing to perform this operation.  

**II - For a particular country c and a particular event E, measure the impact of E on c’s wikipedia characterization.**

« Measuring the impact of E on c’s wikipedia characterization » means :
Looking at the number of edits during E for every page in c’s wikipedia characterization. Compare it the number of edits during a control time interval TI of same duration as E (No major event should happen during TI). 
The point above is actually not clear enough and I’m not sure how to formalize it. 

I couldn’t actually find a recent dataset that provides edits informations. The only one that exists stops in 2008. A solution could be to scrap the data ourselves through an api or directly through the HTML page. 

Conclusion 

Of course some of the arguments I make are not fully justified and could be discussed. The main point is to have this discussion asap so that we don’t get ourselves in an unfeasible project.

**Comment by patryk-oleniuk-epfl** [comment]
In my opinion we should scrap the data from HTML / API, which is of course tiring, but we can obtain the exact data that we want and for very recent data.
For first 3 events, like they said in the comments, we can manually do things, scrap only 1 wikipedia page per event (e.g about the country) and see what can we see there.

For example, for Ukraine website, the current changes are like 4 per day.
But in March 2014, when the war was arising, they were ~10/day. This is already something.
https://en.wikipedia.org/w/index.php?title=Ukraine&offset=20140401000000&limit=250&action=history&tagfilter= 
We can also easily analyse the change in content in Bytes. 

I agree the project is not very well defined and we're not sure that we'll explore something very correlated. But I'm sure we'll discover something new that nobody looked for with the data e.g. above. Even if we only analyse changes for the main country site, this is already enough if we do detailed analysis and we can also choose some sub-sites manually.

But anyway, the other project is also OK but Im not sure is there an easy way to change the project without loosing points.

**Comment by Amaumaury** [comment]
Your approach is probably better for an initial investigation. The only problem is that if there exist a correlation between events and edits this method won't necessarily show it. Anyway, if we try to formalize your idea we can get : 

1 - Pick some major events E related to a country C
2 - Find some datasets that describe those events. Events E are thus defined by multiple sub-events e. 

The goal is then to find evidences of the correlation between events/edits. The fast and dirty way: 

3 - Plot the number of edits over time on C's wikipedia page and mark the sub-events on the timeline. 

We will then have a visual argument that can provide some informal evidence but we'll have to find a rigorous proof for this. 

Do you agree with this ? If yes, we just need to pick the events we want to study. Ukraine war is probably a good idea as we can couple it easily with the UCDP dataset. Furthermore, Ukraine's wikipedia page is very active.

**Comment by patryk-oleniuk-epfl** [comment]
Yes, this is exactly what I meant to do, to just visualize it like you said, especially for the current milestone its OK. Then for the whole project we can consider different things like:
- correlation between events and countries wiki page + maybe try to add some other related wiki in the tree.
- correlation between general country stability and  the changes in its wikipedia. I bet that Switzerland or France have much more stability in their wikipedia than e.g. Ukraine.
- For any country, we can see how many times there are rises in the modifications and try to find "how many time in the year something happened?" (by counting the big rises in changes). Make a map out of it (e.g. divided by size of the country).
-  correlation in English Wikipedia Changes and The National Wikipedia changes, how fast the changes are changed in the English wikipedia after changing the national (just by comparing the peak times).
- how fast wikipedia is changed after the event happened (or maybe even before?)

Theres a lot other small things and ideas that we could explore- im sure some of them will succeed to be at least "kind of correlated and cool". 
I talked with my colleagues from ADA from last year and they said "our idea is pretty cool and innovative, but of course not super straight forward."

**Comment by Amaumaury** [comment]
The problems I can see with in your bullet points

**Point 1** 
How do you want to show that there is a correlation between events and edits ? Visualization are good but certainly not enough. We'll have to find a proper way to show the correlation. 

**Point 2**
Comparing the number of wikipedia changes between any country is possible only if their main wikipage have similar structure. Fortunately it seems to be the case when I checked Spain, France and United States. Still there are some minor differences. 
I think there are others assumptions that we have to make when comparing the edits but there are very subtle and hard to formalize. 

**Point 3**
Why not, we still need to be able to detect those "big changes" which is hard to generalize. My initial idea would be to compute the derivative of our signal (i.e. the number of edits over time) and set a threshold to detect "big changes". However, selecting this threshold would require a bit of hacking skills. 

**Point 4**
It's an interesting question but I don't think it stands in the scope of our project. 

**Point 5**
We could investigate this. However it should be linked to our initial goal. Furthermore, if we want to do something fully rigorous this point is necessary. 

> (or maybe even before?)

There's no way. Intuitively you can only think that there is a causal relationship between events and edits. If edits happen before events there would be something superduper strange. 

I agree with your colleagues, our idea is very exciting. The more I dig, the more I realized how much gold there is in the wikipedia data.
